### PR TITLE
feat: query plan for subscriptions

### PR DESCRIPTION
**Pull request by thisisnithin**
Related https://github.com/wundergraph/cosmo/pull/1425

### CODE REVIEW

File: v2/pkg/engine/resolve/resolve.go [hunk 1]
```diff
@@ -834,6 +834,32 @@ func (r *Resolver) ResolveGraphQLSubscription(ctx *Context, subscription *GraphQ
 		msg := []byte(`{"errors":[{"message":"invalid input"}]}`)
 		return writeFlushComplete(writer, msg)
 	}
+
+	if ctx.ExecutionOptions.SkipLoader {
+		t := newTools(r.options, r.allowedErrorExtensionFields, r.allowedErrorFields)
```
Comment (jensneuse) [inline v2/pkg/engine/resolve/resolve.go:841]: Add a comment here explaining what we're doing and why.

File: v2/pkg/engine/resolve/resolve.go [hunk 2]
```diff
@@ -921,6 +947,32 @@ func (r *Resolver) AsyncResolveGraphQLSubscription(ctx *Context, subscription *G
...
```
Comment (jensneuse) [inline v2/pkg/engine/resolve/resolve.go:956]: Add comment here as well.

File: v2/pkg/engine/resolve/resolve_test.go [hunk 3]
```diff
@@ -5062,6 +5072,35 @@ func TestResolver_ResolveGraphQLSubscription(t *testing.T) {
 		recorder.AwaitComplete(t, defaultTimeout)
 		fakeStream.AwaitIsDone(t, defaultTimeout)
 	})
+
+	t.Run("renders query plan with trigger", func(t *testing.T) {
```
Comment (jensneuse) [inline v2/pkg/engine/resolve/resolve_test.go:5212]: Please add a test that contains both a trigger and at least one additional fetch.

File: v2/pkg/engine/resolve/context.go [hunk 4]
```diff
@@ -14,6 +14,7 @@ import (
 
 type Context struct {
 	ctx              context.Context
+	Query            string
```
Comment (jensneuse) [inline v2/pkg/engine/resolve/context.go]: Try to make it work without having to expose Query and set it from Router.

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: v2/pkg/engine/resolve/resolve.go (hunk 1 above)
Comment (thisisnithin) [reply v2/pkg/engine/resolve/resolve.go:841]: Added

File: v2/pkg/engine/resolve/resolve.go (hunk 2 above)
Comment (thisisnithin) [reply v2/pkg/engine/resolve/resolve.go:956]: Added

File: v2/pkg/engine/resolve/resolve_test.go (hunk 3 above)
Comment (thisisnithin) [reply v2/pkg/engine/resolve/resolve_test.go:5212]: Done

File: v2/pkg/engine/resolve/context.go (hunk 4 above)
Comment (thisisnithin) [reply v2/pkg/engine/resolve/context.go]: Done

File: v2/pkg/engine/resolve/fetchtree.go [hunk 5]
```diff
@@ -165,8 +169,22 @@ func (n *FetchTreeNode) QueryPlan() *FetchTreeQueryPlanNode {
 	if n == nil {
 		return nil
 	}
+
 	plan := n.queryPlan()
 	plan.Version = "1"
+
+	if n.Trigger != nil {
+		f := n.Trigger.Item.Fetch.(*SingleFetch)
```
Comment (jensneuse) [inline v2/pkg/engine/resolve/fetchtree.go]: Missing nil check on `n.Trigger.Item`, `n.Trigger.Item.Fetch`.

Missing type check on *SingleFetch type cast:

```go
if f,ok := n.Trigger.Item.Fetch.(*SingleFetch);ok {
	// do stuff with f
}
```

**Review by jensneuse — CHANGES_REQUESTED** [review]

**Review by jensneuse — APPROVED** [review]

**Pull request merged**